### PR TITLE
Add support for Macbook(2,1)

### PR DESCRIPTION
**Pull request by edrex**
I'm bringing up this old MBP from my bottom drawer. I think it was my second laptop.. It is weird in these ways:

 - 32bit UEFI on a 64bit machine
 - In UEFI mode, Radeon vbios isn't available to the OS (apple grabs it somehow in bootloader)

Tasks:
 - [x] Point to process to bootstrap 64bit OS
 - [x] Install 32bit grub
 - [ ] Build kernel with patch to load vbios in radeon module
 - [ ] Point to process for dumping vbios or otherwise obtaining it.

The vbios thing applies to a bunch of models. The 32 bit uefi is only up to 2007 AFAIK but still several models.

### CODE REVIEW

**Comment by edrex** [comment]
This is still going to be a thing, just need to do a few other things in nix land first.

File: apple/macbook-pro/2-2/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  #  - [ ] put firmware in the right place
+  #        - Load custom Radeon firmware for Macbook Pro / Kernel & Hardware / Arch Linux Forums https://bbs.archlinux.org/viewtopic.php?pid=1810437#p1810437
+  #        - https://discourse.nixos.org/t/trying-to-include-custom-firmware-but-it-doesnt-appear-under-run-current-system/3044
+  boot.kernelParams = ["nomodeset"];
+ 
+      boot.kernelPatches = [
+        { name = "radeon-vbios-from-dump";
+          patch = ./radeon_bios.patch ;
+        }
+      ];
```
Comment (Mic92) [inline apple/macbook-pro/2-2/default.nix:23]: Maybe run nixpkgs-fmt on this file.

File: common/pc/uefi32/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# 32 Bit UEFI
+
+This module forces use of a 32bit EFI GRUB, needed to boot 64bit OSes on systems with a 64-bit CPU but 32-bit EFI firmware, such as Apple's ["Late 2006"](http://refit.sourceforge.net/info/apple_efi.html) hardware.
+
+## Booting 64-bit install media
+Before you can utilize this module, you will need to somehow boot your 64-bit install media, which as-is will fail since firmware will prefer the EFI program over MBR, but it will fail to load because it is 64bit. 
+
+I'm aware of three options:
+
+ - First install a 32-bit OS with a 32-bit EFI bootloader (GRUB) then manually type the boot params for your 64-bit installer into GRUB's console. This is detailed at [Bootloader#Installing x86_64 NixOS on IA-32 UEFI](Installing x86_64 NixOS on IA-32 UEFI)). Importing this module should be sufficient for configuring GRUB correctly on both the initial 32-bit system and the bootstrapped 64-bit one.
```
Comment (edrex) [inline common/pc/uefi32/README.md:10]: fixme: broken link

**Comment by Mic92** [comment]
Still working on this?

**Comment by edrex** [comment]
I had the laptop but not its power supply for a couple months. Just back in
Portland where the power supply is today. The configuration is pretty much
Works For Me now as I recall

**Comment by edrex** [comment]
Oh never mind, there's still the thing with the graphics card. Yeah I'll
get back to it unless someone else picks it up first.

**Comment by Mic92** [comment]
Please re-open if you want to continue to work on this.

**Pull request closed without merging**